### PR TITLE
Network bootstrapper should check for duplicate node names improvements

### PR DESCRIPTION
**Pull request by stanly-johnson**
👮🏻👮🏻👮🏻 !!!! DESCRIBE YOUR CHANGES HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [x] Have you run the unit, integration and smoke tests as described here? https://docs.corda.net/head/testing.html
- [ ] If you added/changed public APIs, did you write/update the JavaDocs?
- [ ] If the changes are of interest to application developers, have you added them to the changelog, and potentially release notes?
- [ ] If you are contributing for the first time, please read the agreement in CONTRIBUTING.md now and add to this Pull Request that you agree to it.

Thanks for your code, it's appreciated! :)

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/network/NetworkBootstrapper.kt [hunk 1]
```diff
@@ -160,6 +166,22 @@ class NetworkBootstrapper {
         }
     }
 
+    /*the function checks for duplicate myLegalName in the all the *_node.conf files
+    All the myLegalName values are added to a HashSet - this helps detect duplicate values.
+    If a duplicate name is found the process is aborted with an error message
+    */
+    private fun checkForDuplicateLegalNames(nodeInfoFiles: List<Path>) {
+      val legalNames = HashSet<String>()
```
Comment (shamsasari) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/network/NetworkBootstrapper.kt]: We use 4 space "tabs", but looking at our coding style guide I realise we don't actually document that! Something for next time.

Comment (stanly-johnson) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/network/NetworkBootstrapper.kt]: Next time 👍

**Comment by joeldudleyr3** [comment]
@stanly-johnson Hi. Could you leave a comment here stating that your contribution is in accordance with the Developer's Certificate of Origin (https://github.com/corda/corda/blob/master/CONTRIBUTING.md#developer-certificate-of-origin)? Thanks!

**Comment by stanly-johnson** [comment]
My contribution is in accordance with the Developer's Certificate of Origin. Thanks

**Comment by joeldudleyr3** [comment]
Thanks :)

**Pull request merged**